test/linalg/diagonal.jl

```diff
@@ -116,6 +123,12 @@ for relty in (Float32, Float64, BigFloat), elty in (relty, Complex{relty})
                 @test Ac_mul_B!(copy(D), copy(b)) ≈ full(D)'*full(b)
             end
 
+            #a few missing mults
+            bd = Bidiagonal(D2)
+            @test A_mul_Bt(D,D2) ≈ full(D)*full(D2).'
```

Comment:
> Is there a way to use `@which` to see what (`*`) will call?

It used to work but then https://github.com/JuliaLang/julia/issues/17469

I think it would be good to change this. `Ax_mul_Bx` are not really meant to be used.